app/models/user.rb

```diff
@@ -1,6 +1,12 @@
 class User < ApplicationRecord
   validates :name, :email, :spire, presence: true
 
+  scope :dispatchers, -> { where.not admin: true }
+
+  def can_delete?(item)
+    admin? || (item.is_a?(LogEntry) && item.user == self)
+  end
```

Comment:
Nah. I mean it's a little weird but the other ways of doing this don't make more sense.